modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -1292,658 +70,77 @@ cv::gimpl::GStreamingExecutor::GStreamingExecutor(std::unique_ptr<ade::Graph> &&
-            }
-            break;
-        case NodeKind::SINK:
-            {
-                const auto sink_idx
-                    = m_gim.metadata(nh).get<Sink>().proto_index;
-                GAPI_Assert(sink_idx < m_sinks.size());
-                m_sinks[sink_idx] = nh;
-
-                // Also initialize Sink's input queue
-                ade::TypedGraph<DataQueue> qgr(*m_island_graph);
-                GAPI_Assert(nh->inEdges().size() == 1u);
-                qgr.metadata(nh->inEdges().front()).set(DataQueue(queue_capacity));
-                m_sink_queues[sink_idx] = qgr.metadata(nh->inEdges().front()).get<DataQueue>().q.get();
-
-                // Assign a desync tag
-                const auto sink_out_nh = gm.metadata().get<Protocol>().out_nhs[sink_idx];
-                if (gm.metadata(sink_out_nh).contains<DesyncPath>()) {
-                    // metadata().get_or<> could make this thing better
-                    m_sink_sync[sink_idx] = gm.metadata(sink_out_nh).get<DesyncPath>().index;
-                }
-            }
-            break;
-        default:
-            GAPI_Assert(false);
-            break;
-        } // switch(kind)
-    } // for(gim nodes)
-
-    // If there are desynchronized parts in the graph, there may be
-    // multiple theads polling every separate (desynchronized)
-    // branch in the graph individually. Prepare a mapping information
-    // for any such thread
-    for (auto &&idx : ade::util::iota(m_sink_queues.size())) {
-        auto  path_id = m_sink_sync[idx];
-        auto &info    = m_collector_map[path_id];
-        info.queues.push_back(m_sink_queues[idx]);
-        info.mapping.push_back(static_cast<int>(idx));
-    }
-
-    // Reserve space in the final queue based on the number
-    // of desync parts (they can generate output individually
-    // per the same input frame, so the output traffic multiplies)
-    GAPI_Assert(m_collector_map.size() > 0u);
-    m_out_queue.set_capacity(queue_capacity * m_collector_map.size());
-
-    // FIXME: The code duplicates logic of collectGraphInfo()
-    cv::gimpl::GModel::ConstGraph cgr(*m_orig_graph);
-    auto meta = cgr.metadata().get<cv::gimpl::Protocol>().out_nhs;
-    out_info.reserve(meta.size());
-
-    ade::util::transform(meta, std::back_inserter(out_info), [&cgr](const ade::NodeHandle& nh) {
-        const auto& data = cgr.metadata(nh).get<cv::gimpl::Data>();
-        return cv::GTypeInfo{data.shape, data.kind, data.ctor};
-    });
+    , m_gim(*m_island_graph) {
 }
 
-cv::gimpl::GStreamingExecutor::~GStreamingExecutor()
-{
+cv::gimpl::GStreamingExecutor::~GStreamingExecutor() {
     // FIXME: this is a temporary try-catch exception hadling.
-    // Need to eliminate throwings from stop()
+    // Need to eliminate throwings from stop(): make stop_impl() & wait_shutdown() no_throw
     try {
-        if (state == State::READY || state == State::RUNNING)
+        if (state == State::READY || state == State::RUNNING) {
```

Comment:
Shouldn't we make `state` an `atomic`/guard it? I guess the question is not related to this PR but still

Reply:
Please, ignore if it's not a critical section